DSpace/DSpace
# Fix for #8918 The Community Administrator should not be able to view all communities/collections in the create/edit community and collection sections.

# References
* Fixes #8918 

# Description
The Community Administrator should not be able to view all communities/collections in the create/edit community and collection sections.

# Instructions for Reviewers
Currently, users can see all communities/collections in the create/edit community and collection sections. However, the expected behavior is that only the communities/collections where the user is a community or collection admin should be displayed.

# List of changes in this PR:
•	Added a method retrieveObjectWithAdmin in the AuthorizeService and implemented it in the AuthorizedServiceImp class. This method returns a filtered result, showing only the communities/collections where the user is a community or collection admin.

# How to Test
1.	Make a user the admin of a few communities.
2.	Navigate to the “New” section and click on "Community" or “Collection”. Only the communities where the user is community admin should be displayed.
3.	For the Edit section:
o	Go to Edit > click “Community”. Again, only the communities where the user is an admin should be shown.
o	For collections, go to Edit > click “Collection”. Only the collections where the user is a collection or community admin should be listed. If the user is a community admin for a community, all collections within that community should also be shown.

Bug Reproduction Steps
1.	New Section:
o	Log in as a community admin, go to New > Community. Currently, all communities are shown in the list.
o	Similarly, for collections, go to the New section > click Collection. Here, all collections are shown in the list.
2.	Edit Section:
o	Log in as a community admin, go to Edit > Community. Currently, all communities are shown in the list.
o	Similarly, for collections, go to Edit > click Collection. Here, all collections are shown in the list.
